unlock-app/src/middlewares/walletMiddleware.js

```diff
@@ -129,29 +124,7 @@ export default function walletMiddleware({ getState, dispatch }) {
         walletService.connect(action.provider)
       } else if (action.type === CREATE_LOCK && action.lock.address) {
         ensureReadyBefore(() => {
-          walletService
-            .createLock(action.lock, getState().account.address)
-            .then(() => {
-              if (config.services.storage) {
-                generateSignature(
-                  walletService.web3,
-                  getState().account.address,
-                  action.lock
-                )
-                  .then(token => {
-                    dispatch(
-                      storeLockName(
-                        getState().account.address,
-                        token.data,
-                        token.result
-                      )
-                    )
-                  })
-                  .catch(error => {
-                    dispatch(signatureError(error))
-                  })
-              }
-            })
+          walletService.createLock(action.lock, getState().account.address)
```

Comment:
I'll have to look at the current implementation of createLock but I think this might introduce a change in behavior we may not be intending. 

I previously stumbled upon this, a rejected promise is returned from createLock in a few scenarios, one of which is something like a person canceling the the lock creation in metamask. To cover this case we'll need to ensure that request to sign is dispatched only upon success

Reply:
Good point. I think though that the request to sign now only happens when/if the lock has been created? (it's been moved to the `storageMiddleware` (see `action.type === CREATE_LOCK && action.lock.address`).

Reply:
Ah that makes sense 👍